DAOS-14334 control: Fix PoolCreateResp.leader

liw opened the pull request:
The dmg pool create command currently outputs invalid service leaders. For instance, with master:

    Service Leader       : 0
    Service Ranks        : [1-5]

This patch fixes the type conversion from mgmtpb.PoolCreateResp to control.PoolCreateResp.

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate owners.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

daosbuild1 commented:
Test stage Test RPMs on EL 8.6 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-13850/1/execution/node/1157/log

daosbuild1 commented:
Test stage Unit Test on EL 8.8 completed with status UNSTABLE.  https://build.hpdd.intel.com/job/daos-stack/job/daos//view/change-requests/job/PR-13850/3/testReport/

daosbuild1 commented:
Test stage Functional Hardware Large completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-13850/3/display/redirect

daosbuild1 commented:
Test stage Functional Hardware Medium completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-13850/3/display/redirect

daosbuild1 commented:
Test stage Unit Test on EL 8.8 completed with status UNSTABLE.  https://build.hpdd.intel.com/job/daos-stack/job/daos//view/change-requests/job/PR-13850/4/testReport/

kjacque commented on src/control/lib/control/pool.go:
```diff
@@ -222,7 +222,7 @@ type (
 	PoolCreateResp struct {
 		UUID      string   `json:"uuid"`
-		Leader    uint32   `json:"svc_ldr"`
+		SvcLdr    uint32   `json:"svc_ldr"`
```
I think you could leave this name as-is, since it's already translated to the correct JSON.

kjacque reviewed (approved):
LGTM

daosbuild1 commented:
Test stage Functional Hardware Medium Verbs Provider completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-13850/5/execution/node/1540/log

liw commented:
daos_test/suite.py: rebuild_simple: DAOS-15431

tanabarr reviewed (approved):
Took awhile to figure out why these changes fix the issue. The type conversion between protobuf and native of PoolCreateResp was broken because the JSON tag was wrong on the native type `Leader    uint32   `json:"svc_ldr"` as the pb definition was `uint32 leader`. Thanks for fixing.

liw commented:
> Took awhile to figure out why these changes fix the issue. The type conversion between protobuf and native of PoolCreateResp was broken because the JSON tag was wrong on the native type "Leader uint32 `json:"svc_ldr"`" as the pb definition was `uint32 leader`. Thanks for fixing.

@tanabarr, it is hard to see, indeed. I've added a mention to the JSON tag in the PR description. It _was_ easy to tell in the initial version of the PR where we changed the JSON tag, but due to `dmg` command compatibility considerations, we switched the whole approach to changing the Protobuf definition, and I should have beefed up the PR description. Sorry for the confusion.

daosbuild1 commented:
Test stage Functional Hardware Medium Verbs Provider completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-13850/6/execution/node/1539/log

liw commented:
daos_test/suite.py: rebuild_simple: DAOS-15431
